data_juicer/ops/mapper/generate_instruction_mapper.py

```diff
@@ -0,0 +1,247 @@
+
+
+@UNFORKABLE.register_module(OP_NAME)
+@OPERATORS.register_module(OP_NAME)
+class GenerateInstructionMapper(Mapper):
+    """Mapper to generate new instruction text data.
+    You should configure an empty dataset in your yaml config file:
+    ```
+    dataset_config:
+      type: 'EmptyFormatter'
+      length: ${The number of generated samples}
+    ```
+    The number of samples generated is determined by
+    the length of the empty dataset.
+    """
+    _accelerator = 'cuda'
+
+    def __init__(self,
+                 hf_model,
+                 seed_file,
+                 instruct_num,
+                 similarity_threshold: float = 0.7,
+                 prompt_template: str = None,
+                 enable_vllm: bool = False,
+                 tensor_parallel_size: int = None,
+                 max_model_len: int = None,
+                 max_num_seqs: int = 256,
+                 sampling_params: Dict = {},
+                 *args,
+                 **kwargs):
+        """
+        Initialization method.
+
+        :param hf_model: Hugginface model id.
+        :param seed_file: Seed file path, chatml format.
+        :param instruct_num: The number of instruction samples.
+            Randomly select N samples from "seed_file" and
+            put them into prompt as instruction samples.
+        :param generate_num: The number of generated samples.
+        :param similarity_threshold: The similarity score threshold
+            between the generated samples and the seed samples.
+            Range from 0 to 1. Samples with similarity score less than
+            this threshold will be kept.
+        :param prompt_template: Prompt template for generate samples.
+            Please make sure the template contains "{augmented_data}",
+            which corresponds to the augmented samples.
+        :param enable_vllm: Whether to use vllm for inference acceleration.
+        :param tensor_parallel_size: It is only valid when enable_vllm is True.
+            The number of GPUs to use for distributed execution with tensor
+            parallelism.
+        :param max_model_len: It is only valid when enable_vllm is True.
+            Model context length. If unspecified, will be automatically
+            derived from the model config.
+        :param max_num_seqs: It is only valid when enable_vllm is True.
+            Maximum number of sequences to be processed in a single iteration.
+        :param sampling_params: Sampling parameters for text generation.
+            e.g {'temperature': 0.9, 'top_p': 0.95}
+        :param args: extra args
+        :param kwargs: extra args
+        """
+        super().__init__(*args, num_proc=1, **kwargs)
+
+        self.instruct_num = instruct_num
+        self.similarity_threshold = similarity_threshold
+        self.similarity_type = 'rouge_l'
+        # self._batched_op = True
+
+        if prompt_template is None:
+            prompt_template = DEFAULT_PROMPT_TEMPLATE
```

Comment:
It would be better to make other constants (in lines 23~26) also configurable for broader scenarios and languages?